src/MediatR/Internal/RequestHandlerWrapper.cs

```diff
@@ -48,7 +48,7 @@ internal class RequestHandlerWrapperImpl<TRequest, TResponse> : RequestHandlerWr
             return Handle((IRequest<TResponse>)request, cancellationToken, serviceFactory)
                 .ContinueWith(t =>
                 {
-                    if (t.IsFaulted)
+                    if (t.IsFaulted && t.Exception?.InnerException is not null)
```

Comment:
Would this break functionality - as in, does `ExceptionDispatchInfo` still throw even if the innerEx is null?

Reply:
Yes it does, https://docs.microsoft.com/en-us/dotnet/api/system.runtime.exceptionservices.exceptiondispatchinfo.capture?view=netcore-3.1#exceptions